core/chains/evm/logpoller/orm.go

```diff
@@ -299,66 +312,75 @@ func (o *DbORM) DeleteExpiredLogs(limit int64, qopts ...pg.QOpt) (int64, error)
 			) r ON l.evm_chain_id = $1 AND l.address = r.address AND l.event_sig = r.event
 			AND l.block_timestamp <= STATEMENT_TIMESTAMP() - (r.retention / 10^9 * interval '1 second')
 			LIMIT $2
-		)`,
-			ubig.New(o.chainID), limit)
-	}
-
-	return q.ExecQWithRowsAffected(`WITH r AS
+		)`, ubig.New(o.chainID), limit)
+	} else {
+		result, err = o.db.ExecContext(ctx, `WITH r AS
 		( SELECT address, event, MAX(retention) AS retention
 			FROM evm.log_poller_filters WHERE evm_chain_id=$1 
 			GROUP BY evm_chain_id,address, event HAVING NOT 0 = ANY(ARRAY_AGG(retention))
 		) DELETE FROM evm.logs l USING r
 			WHERE l.evm_chain_id = $1 AND l.address=r.address AND l.event_sig=r.event
 			AND l.block_timestamp <= STATEMENT_TIMESTAMP() - (r.retention / 10^9 * interval '1 second')`, // retention is in nanoseconds (time.Duration aka BIGINT)
-		ubig.New(o.chainID))
+			ubig.New(o.chainID))
+	}
+
+	rowsAffected, _ := result.RowsAffected()
+	return rowsAffected, err
 }
 
 // InsertLogs is idempotent to support replays.
-func (o *DbORM) InsertLogs(logs []Log, qopts ...pg.QOpt) error {
+func (o *orm) InsertLogs(ctx context.Context, logs []Log) error {
 	if err := o.validateLogs(logs); err != nil {
 		return err
 	}
-
-	return o.q.WithOpts(qopts...).Transaction(func(tx pg.Queryer) error {
-		return o.insertLogsWithinTx(logs, tx)
+	return o.Transaction(ctx, func(orm *orm) error {
+		return o.insertLogsWithinTx(ctx, logs, orm.db.(*sqlx.Tx))
 	})
 }
 
-func (o *DbORM) InsertLogsWithBlock(logs []Log, block LogPollerBlock, qopts ...pg.QOpt) error {
+func (o *orm) InsertLogsWithBlock(ctx context.Context, logs []Log, block LogPollerBlock) error {
 	// Optimization, don't open TX when there is only a block to be persisted
 	if len(logs) == 0 {
-		return o.InsertBlock(block.BlockHash, block.BlockNumber, block.BlockTimestamp, block.FinalizedBlockNumber, qopts...)
+		return o.InsertBlock(ctx, block.BlockHash, block.BlockNumber, block.BlockTimestamp, block.FinalizedBlockNumber)
 	}
 
 	if err := o.validateLogs(logs); err != nil {
 		return err
 	}
 
 	// Block and logs goes with the same TX to ensure atomicity
-	return o.q.WithOpts(qopts...).Transaction(func(tx pg.Queryer) error {
-		if err := o.InsertBlock(block.BlockHash, block.BlockNumber, block.BlockTimestamp, block.FinalizedBlockNumber, pg.WithQueryer(tx)); err != nil {
```

Comment:
Was this a "bug" in the log poller?

Reply:
Yeah it should have been using the tx to insert.